Src/FluentAssertions/Primitives/StringEqualityValidator.cs

```diff
@@ -27,14 +27,51 @@ protected override bool ValidateAgainstSuperfluousWhitespace()
 
         protected override bool ValidateAgainstLengthDifferences()
         {
+            // Logic is a little bit convoluted because I want to avoid calculation
+            // of mismatch segment in case of equalLength == true for performance reason.
+            // If lazy version of FailWith would be introduced, calculation of mismatch
+            // segment can be moved directly to FailWith's argument
+            bool equalLength = subject.Length == expected.Length;
+
+            string mismatchSegment = GetMismatchSegmentForStringsOfDifferentLengths(equalLength);
```

Comment:
🔧 You could prevent that call, but chaining it after the `FailWith` below. See the [CollectionAssertions](https://github.com/fluentassertions/fluentassertions/blob/179fe62b67dfabc4e38c34e9f62afe2d2ead91c8/Src/FluentAssertions/Collections/CollectionAssertions.cs#L1430) for an example

Reply:
Could you elaborate more on this one? I think I need more guidance. 
Best I could come up with is the code below, but it doesn't even compile.
```
.Given(() => subject)
.ForCondition(x => subject.Length == expected.Length)
.FailWith(
   ExpectationDescription + "{0} with a length of {1}{reason}, but {2} has a length of {3}, differs near {4}.",
   (_) => expected, (_) => expected.Length, (x) => x, (x) => x.Length, (_) => mismatchSegment)
 .SourceSucceeded; // this line fails 
```
Even if the last list compiled it still feels like a "workaround" for the fact that "lazy" version of FailWith is not available on ```AssertionScope```, doesn't it?

Reply:
Yeah, I got your approach wrong. You want to combine everything in one failure message.